### PR TITLE
Refactored CIM export to bytesIO

### PR DESCRIPTION
**Pull request by gpoderys**

### CODE REVIEW

**Comment by Haigutus** [comment]
Implements https://github.com/Baltic-RCC/EMF/issues/9
@gpoderys please add link to the ticket when committing

File: emf/loadflow_tool/helper.py [hunk 1]
```diff
@@ -309,4 +309,20 @@ def get_metadata_from_filename(file_name):
             file_metadata["Model.domain"],\
             file_metadata["Model.forEntity"] = entity_and_area_list
 
-    return file_metadata
\ No newline at end of file
+    return file_metadata
+
+
+def export_model(network: pypowsybl.network, CGM_meta):
```
Comment (Haigutus) [inline emf/loadflow_tool/helper.py:315]: profiles should be a list of profile string ala ["EQ", "SSH"] on the function call. Default in function call should None, as listis are not good if deified in as default function call. Replace none with list of "SV,SSH,TP,EQ"

File: emf/loadflow_tool/helper.py [hunk 2]
```diff
@@ -309,4 +309,20 @@ def get_metadata_from_filename(file_name):
...
+    bytes_object = network.save_to_binary_buffer(
+        format="CGMES",
+        parameters={
+            "iidm.export.cgmes.modeling-authority-set": CGM_meta['pmd:modelingAuthoritySet'],
+            "iidm.export.cgmes.base-name": filename_from_metadata(CGM_meta).split(".xml")[0],
+            "iidm.export.cgmes.profiles": "SV,SSH,TP,EQ",
+            "iidm.export.cgmes.naming-strategy": "cgmes",  # identity, cgmes, cgmes-fix-all-invalid-ids
+        })
+    temp_dir = ""
+    bytes_object.name =os.path.join(temp_dir, f"MERGED_MODEL_{uuid.uuid4()}.zip")
```
Comment (Haigutus) [inline emf/loadflow_tool/helper.py:326]: file_base_name = filename_from_metadata(CGM_meta).split(".xml")[0]
bytes_object.name =f"{file_base_name}_{uuid.uuid4()}.zip"

File: emf/loadflow_tool/helper.py [hunk 3]
```diff
@@ -309,4 +309,20 @@ def get_metadata_from_filename(file_name):
...
+
+def export_model(network: pypowsybl.network, CGM_meta):
+
+    bytes_object = network.save_to_binary_buffer(
+        format="CGMES",
+        parameters={
+            "iidm.export.cgmes.modeling-authority-set": CGM_meta['pmd:modelingAuthoritySet'],
+            "iidm.export.cgmes.base-name": filename_from_metadata(CGM_meta).split(".xml")[0],
+            "iidm.export.cgmes.profiles": "SV,SSH,TP,EQ",
+            "iidm.export.cgmes.naming-strategy": "cgmes",  # identity, cgmes, cgmes-fix-all-invalid-ids
```
Comment (Haigutus) [inline emf/loadflow_tool/helper.py:323]: Investigate the difference between "cgmes" and "cgmes-fix-all-invalid-ids"

File: emf/loadflow_tool/merge.py [hunk 4]
```diff
@@ -73,58 +73,47 @@
...
-merged_model["NETWORK"].dump(export_file_path,
-                           format="CGMES",
-                           parameters={
-                                "iidm.export.cgmes.modeling-authority-set": CGM_meta['opdm:OPDMObject']['pmd:modelingAuthoritySet'],
-                                "iidm.export.cgmes.base-name": filename_from_metadata(CGM_meta['opdm:OPDMObject']).split("_SV")[0],
-                                "iidm.export.cgmes.profiles": "SV",
-                                "iidm.export.cgmes.naming-strategy": "cgmes",  # identity, cgmes, cgmes-fix-all-invalid-ids
-                                       })
-
+exported_model = export_model(merged_model["NETWORK"])
```
Comment (Haigutus) [inline emf/loadflow_tool/merge.py:100]: Export only SV here

File: emf/loadflow_tool/merge.py [hunk 5]
```diff
@@ -73,58 +73,47 @@
...
+logger.info(f"Exporting merged model to {exported_model.name}")
 
 # Load SV data
-sv_data = pandas.read_RDF([export_file_path])
+sv_data = pandas.read_RDF([exported_model.name])
```
Comment (Haigutus) [inline emf/loadflow_tool/merge.py:104]: just exported_model should work, .name gives path, but read_RDF can load in bytesIO directly

File: emf/loadflow_tool/merge_process_poc.py [hunk 6]
```diff
@@ -113,42 +113,33 @@
...
 export_report = pypowsybl.report.Reporter()
-merged_model["NETWORK"].dump(export_file_path,
-                           format="CGMES",
-                           parameters={
-                                "iidm.export.cgmes.modeling-authority-set": CGM_meta['opdm:OPDMObject']['pmd:modelingAuthoritySet'],
-                                "iidm.export.cgmes.base-name": filename_from_metadata(CGM_meta['opdm:OPDMObject']).split("_SV")[0],
-                                "iidm.export.cgmes.profiles": "SV",
-                                "iidm.export.cgmes.naming-strategy": "cgmes",  # identity, cgmes, cgmes-fix-all-invalid-ids
-                                       })
+exported_model = export_model(merged_model["NETWORK"])
```
Comment (Haigutus) [inline emf/loadflow_tool/merge_process_poc.py:141]: Export only SV here

File: emf/loadflow_tool/helper.py [hunk 7]
```diff
@@ -309,4 +310,26 @@ def get_metadata_from_filename(file_name):
             file_metadata["Model.domain"],\
             file_metadata["Model.forEntity"] = entity_and_area_list
 
-    return file_metadata
\ No newline at end of file
+    return file_metadata
+
+
+def export_model(network: pypowsybl.network, CGM_meta, profiles=None):
```
Comment (Haigutus) [inline emf/loadflow_tool/helper.py:316]: 1. Rename CGM_meta to opdm_object_meta
2. Get base filename only once, in the beginning of the function

File: emf/loadflow_tool/helper.py (hunk 3 above)
Comment (gpoderys) [reply emf/loadflow_tool/helper.py:331]: image

Comment (Haigutus) [reply emf/loadflow_tool/helper.py:331]: we should change it to the fix-all-invalid-ids, to reduce possible errors at OPDE publication

**Pull request merged**